modules/utxo-lib/src/bitgo/transaction.ts

```diff
@@ -9,32 +9,44 @@ import { DashTransactionBuilder } from './dash/DashTransactionBuilder';
 import { ZcashTransactionBuilder } from './zcash/ZcashTransactionBuilder';
 import { ZcashNetwork, ZcashTransaction } from './zcash/ZcashTransaction';
 
-export function createTransactionFromBuffer(
+export function createTransactionFromBuffer<TNumber extends number | bigint = number>(
   buf: Buffer,
   network: Network,
+  amountType: 'number' | 'bigint' = 'number',
   { version }: { version?: number } = {}
-): UtxoTransaction {
+): UtxoTransaction<TNumber> {
+  if (amountType !== 'number' && (getMainnet(network) === networks.dash || getMainnet(network) === networks.zcash)) {
+    throw new Error('dash and zcash must use number amount type; bigint amount type is recommended for doge only');
+  }
   switch (getMainnet(network)) {
     case networks.bitcoin:
     case networks.bitcoincash:
     case networks.bitcoinsv:
     case networks.bitcoingold:
-    case networks.litecoin:
     case networks.dogecoin:
-      return UtxoTransaction.fromBuffer(buf, false, network);
+    case networks.litecoin:
+      return UtxoTransaction.fromBuffer<TNumber>(buf, false, amountType, network);
     case networks.dash:
-      return DashTransaction.fromBuffer(buf, false, network);
+      return DashTransaction.fromBufferDash(buf, false, network) as unknown as UtxoTransaction<TNumber>;
     case networks.zcash:
-      return ZcashTransaction.fromBufferWithVersion(buf, network as ZcashNetwork, version);
+      return ZcashTransaction.fromBufferWithVersion(
+        buf,
+        network as ZcashNetwork,
+        version
+      ) as unknown as UtxoTransaction<TNumber>;
   }
 
   /* istanbul ignore next */
   throw new Error(`invalid network`);
 }
 
 /* istanbul ignore next */
-export function createTransactionFromHex(hex: string, network: Network): UtxoTransaction {
-  return createTransactionFromBuffer(Buffer.from(hex, 'hex'), network);
+export function createTransactionFromHex<TNumber extends number | bigint = number>(
+  hex: string,
+  network: Network,
+  amountType: 'number' | 'bigint' = 'number'
```

Comment:
Change in required parameters. Please record as BREAKING CHANGE in commits.

Reply:
See first conversation